### PR TITLE
imx93 pinctrl/misc update

### PR DESCRIPTION
**Pull request by CkovMk**
Brief:
Update pinctrl to match https://github.com/zephyrproject-rtos/hal_nxp/pull/273. 
Change clock driver to use mcux_ccm_rev2.
Increase MMU region count to 64.
Enable lpuart1.

(I know there's mcux_ccm_rev3 under discussion, but my work was based on mcux_ccm_rev2)

### CODE REVIEW

File: boards/arm64/mimx93_evk/mimx93_evk-pinctrl.dtsi [hunk 1]
```diff
@@ -9,8 +9,8 @@
 &pinctrl {
 	uart2_default: uart2_default {
 		group0 {
-			pinmux = <&iomuxc_uart2_rxd_uart_rx_uart2_rx>,
-				<&iomuxc_uart2_txd_uart_tx_uart2_tx>;
+			pinmux = <&iomuxc1_uart2_rxd_lpuart_rx_lpuart2_rx>,
```
Comment (danieldegrasse) [inline boards/arm64/mimx93_evk/mimx93_evk-pinctrl.dtsi:22]: Please add a commit updating `west.yml` to reference https://github.com/zephyrproject-rtos/hal_nxp/pull/273, since you are updating the pin control node names here.

File: drivers/clock_control/clock_control_mcux_ccm_rev2.c [hunk 2]
```diff
@@ -114,6 +114,23 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 	return 0;
 }
 
+static int mcux_ccm_init(const struct device *dev)
+{
+#ifdef CONFIG_SOC_MIMX93_A55
```
Comment (danieldegrasse) [inline drivers/clock_control/clock_control_mcux_ccm_rev2.c:119]: I'm unclear why this change is needed- is this variable declared in the HAL?

**Comment by mmahadevan108** [comment]
cc @JiafeiPan

File: drivers/clock_control/clock_control_mcux_ccm_rev2.c [hunk 3]
```diff
@@ -106,8 +106,28 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
...
+	g_clockSourceFreq[kCLOCK_Osc24M] 			= 24000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1]			= 4000000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd0]		= 1000000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd0Div2]	= 500000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd1]		= 800000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd1Div2]	= 400000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd2]		= 625000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd2Div2]	= 312500000U;
+	g_clockSourceFreq[kCLOCK_AudioPll1]			= 393216000U;
+	g_clockSourceFreq[kCLOCK_AudioPll1Out]		= 393216000U;
```
Comment (JiafeiPan) [inline drivers/clock_control/clock_control_mcux_ccm_rev2.c:129]: As we discussed, these clock source frequency need to be initialized in hal driver, we already have such fix, so port the fix to hal_nxp firstly.

File: boards/arm64/mimx93_evk/mimx93_evk_a55.dts [hunk 4]
```diff
@@ -30,6 +30,14 @@
 	};
 };
 
+&lpuart1 {
+	status = "okay";
```
Comment (JiafeiPan) [inline boards/arm64/mimx93_evk/mimx93_evk_a55.dts:34]: Don't enable lpuart1 by default, you can enable it in project's overlay dts if you need to use it.

File: drivers/clock_control/clock_control_mcux_ccm_rev2.c (hunk 2 above)
Comment (JiafeiPan) [reply drivers/clock_control/clock_control_mcux_ccm_rev2.c:119]: These clock source frequency need to be initialized in hal driver, we already have patch in mcux-sdk's internal development repo, so push it to github repo and port the fix to hal_nxp firstly before we merge this PR.

File: drivers/clock_control/clock_control_mcux_ccm_rev2.c [hunk 5]
```diff
@@ -106,8 +106,28 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 	default:
 		return -EINVAL;
 	}
-
+#ifdef CONFIG_SOC_MIMX93_A55
```
Comment (JiafeiPan) [inline drivers/clock_control/clock_control_mcux_ccm_rev2.c:109]: As we want to use ccm rev2 driver for i.mx93, let's also revert the update in ccm rev1 driver.

**Review by JiafeiPan — CHANGES_REQUESTED** [review]

File: boards/arm64/mimx93_evk/mimx93_evk-pinctrl.dtsi (hunk 1 above)
Comment (JiafeiPan) [reply boards/arm64/mimx93_evk/mimx93_evk-pinctrl.dtsi:22]: As https://github.com/zephyrproject-rtos/hal_nxp/pull/273 depends on this commit, you can create a seperate PR for this patch in order to make thing simple.

File: drivers/clock_control/clock_control_mcux_ccm_rev2.c (hunk 5 above)
Comment (CkovMk) [reply drivers/clock_control/clock_control_mcux_ccm_rev2.c:109]: done.

File: boards/arm64/mimx93_evk/mimx93_evk_a55.dts (hunk 4 above)
Comment (CkovMk) [reply boards/arm64/mimx93_evk/mimx93_evk_a55.dts]: done.

File: drivers/clock_control/clock_control_mcux_ccm_rev2.c (hunk 2 above)
Comment (CkovMk) [reply drivers/clock_control/clock_control_mcux_ccm_rev2.c]: I've port it to hal_nxp. see pr: 
https://github.com/zephyrproject-rtos/hal_nxp/pull/286

File: drivers/clock_control/clock_control_mcux_ccm_rev2.c (hunk 3 above)
Comment (CkovMk) [reply drivers/clock_control/clock_control_mcux_ccm_rev2.c]: ported. 
https://github.com/zephyrproject-rtos/hal_nxp/pull/286

**Comment by danieldegrasse** [comment]
@CkovMk a few comments:
- please don't create new commits when you address review requests. Instead, edit exist commits via a rebase
- Since this PR seems to depend on two separate NXP HAL PRs, I would consider splitting these changes into 2 PRs, one for pin control and one for clock management

**Comment by CkovMk** [comment]
Hi @danieldegrasse , I've moved ccm related changes to #63024 .

> please don't create new commits when you address review requests. Instead, edit exist commits via a rebase
I will use rebase instead.

**Comment by mmahadevan108** [comment]
@CkovMk HAL PR has been merged. Please update `west.yml` to point to the SHA in HAL

**Comment by mmahadevan108** [comment]
@CkovMk Please also address the CI failures.

**Comment by CkovMk** [comment]
Hi @mmahadevan108 , CI build is mostly pass, except one type error in debug format string. I assume it's due to CI is more stict in syntax. See the build failure below:

> In file included from /__w/zephyr/zephyr/include/zephyr/logging/log.h:11,
                 from /__w/zephyr/zephyr/drivers/sensor/default_rtio_sensor.c:11:
/__w/zephyr/zephyr/drivers/sensor/default_rtio_sensor.c: In function 'sensor_submit_fallback':
/__w/zephyr/zephyr/drivers/sensor/default_rtio_sensor.c:238:17: error: format '%zu' expects argument of type 'size_t', but argument 2 has type 'uint32_t' {aka 'unsigned int'} [-Werror=format=]
  238 |         LOG_DBG("Total channels in header: %zu", header->num_channels);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~
      |                                                        |
      |                                                        uint32_t {aka unsigned int}
/__w/zephyr/zephyr/include/zephyr/logging/log_core.h:245:42: note: in definition of macro 'Z_LOG2'
  245 |                 z_log_printf_arg_checker(__VA_ARGS__); \
      |                                          ^~~~~~~~~~~
/__w/zephyr/zephyr/include/zephyr/logging/log.h:73:25: note: in expansion of macro 'Z_LOG'
   73 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/__w/zephyr/zephyr/drivers/sensor/default_rtio_sensor.c:238:9: note: in expansion of macro 'LOG_DBG'
  238 |         LOG_DBG("Total channels in header: %zu", header->num_channels);
      |         ^~~~~~~
/__w/zephyr/zephyr/drivers/sensor/default_rtio_sensor.c:238:46: note: format string is defined here
  238 |         LOG_DBG("Total channels in header: %zu", header->num_channels);
      |                                            ~~^
      |                                              |
      |                                              long unsigned int
      |                                            %u
cc1: all warnings being treated as errors

I don't think here's the right place to fix it...
what do you think?

**Review by decsny — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@CkovMk, please update `west.yml`

**Comment by CkovMk** [comment]
Hi @mmahadevan108 ,
west.yml updated. suggest to merge #63024 first, since they use the same hal/nxp commit hash.

**Review by JiafeiPan — APPROVED** [review]
@dleach02  , could you please review and merge this this PR and https://github.com/zephyrproject-rtos/zephyr/pull/63024?  there is build issue for the board mimx93_evk_a55_sof and mimx93_evk_a55 caused by update from hal_nxp:
https://github.com/zephyrproject-rtos/hal_nxp/pull/273
thanks.

**Review by dleach02 — DISMISSED** [review]

File: west.yml [hunk 6]
```diff
@@ -193,7 +193,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: d5e5358e56542b94bc65b6483396f50ed8f3172d
+      revision: 78d1912dbc2b5f6e114d7fd19a31053716a3b01d
```
Comment (mmahadevan108) [inline west.yml:196]: Please update the hash to pull in the appropriate commit and not the tip of hal. 
Please pull in `73620197038d7ba80fb1f9abd001828b9dd6a27e`

**Review by mmahadevan108 — DISMISSED** [review]
Please update the hash in `west.yml`

**Comment by mmahadevan108** [comment]
Please rebase and resolve merge conflict

**Comment by mmahadevan108** [comment]
@CkovMk , I attempted to resolve the merge conflict via the github UI and maybe have messed up your branch. Can you please force push your local branch.

**Comment by CkovMk** [comment]
Hi @mmahadevan108 , rebased onto main to solve the manifest conflict.

**Review by dleach02 — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]
